node/external/logs/logsRepository.go

```diff
@@ -17,6 +19,10 @@ type logsRepository struct {
 
 func newLogsRepository(storageService dataRetriever.StorageService, marshaller marshal.Marshalizer) *logsRepository {
 	storer := storageService.GetStorer(dataRetriever.TxLogsUnit)
+	if check.IfNil(storer) {
```

Comment:
instead of this approach, I would make the `storageService.GetStorer(dataRetriever.TxLogsUnit)` call return always a not-nil pointer, on all conditions. So, in StorageServiceFactory.setupLogsAndEventsStorer function, on L556 I would add a disabled storer.
This will simplify all involved code and this new if branch will no longer be necessary. 
For a next PR: we might return an error on the GetStorer function whenever we've provided a key that does not contain a valid storer as to align the code to the rest of the project.

Reply:
pushed